frontend/src/employee-frontend/components/child-information/VasuAndLeops.tsx

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { Loading } from 'lib-common/api'
+import Loader from 'lib-components/atoms/Loader'
+import React, { useContext, useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { AddButtonRow } from '../../../lib-components/atoms/buttons/AddButton'
+import { StaticChip } from '../../../lib-components/atoms/Chip'
+import { CollapsibleContentArea } from '../../../lib-components/layout/Container'
+import {
+  Table,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr
+} from '../../../lib-components/layout/Table'
+import { H2 } from '../../../lib-components/typography'
+import colors from '../../../lib-customizations/common'
+import { getVasuDocumentSummary } from '../../api/vasu'
+import { ChildContext } from '../../state'
+import { useTranslation } from '../../state/i18n'
+import { UUID } from '../../types'
+import { VasuDocumentState, VasuDocumentSummary } from '../../types/vasu'
+import { formatDate } from '../../utils/date'
+
+const StateCell = styled(Td)`
+  display: flex;
+  justify-content: flex-end;
+`
+
+const chipColors: Record<VasuDocumentState, string> = {
+  DRAFT: colors.accents.emerald,
+  CREATED: colors.accents.violet,
+  REVIEWED: colors.primary,
+  CLOSED: colors.greyscale.medium
+}
+
+type ChipLabels = Record<VasuDocumentState, string>
+
+interface StateChipProps {
+  state: VasuDocumentState
+  labels: ChipLabels
+}
+
+function StateChip({ labels, state }: StateChipProps) {
+  return <StaticChip color={chipColors[state]}>{labels[state]}</StaticChip>
+}
+
+interface Props {
+  id: UUID
+  startOpen: boolean
+}
+
+const VasuAndLeops = React.memo(function VasuAndLeops({
+  id,
+  startOpen
+}: Props) {
+  const { i18n } = useTranslation()
+  const { vasus, setVasus } = useContext(ChildContext)
+
+  const [open, setOpen] = useState(startOpen)
+
+  useEffect(() => {
+    setVasus(Loading.of())
+    void getVasuDocumentSummary(id).then((response) => {
+      setVasus(response)
+    })
+  }, [id, setVasus])
```

Comment:
might be better as
```
const loadVasus = useRestApi(getVasuDocumentSummary, setVasus)
useEffect(() => loadVasus(id), [id, loadVasus])
```

Reply:
Aye, absolutely. This was a bad copy-paste 😬